### PR TITLE
fix: workspaces amount limits checking

### PR DESCRIPTION
**Pull request by mshaposhnik**
### What does this PR do?
Fix workspaces amount checking when creating from devfile

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/20187

### How to test this PR?

Configure Che:
```
customCheProperties:
      CHE_LIMITS_USER_WORKSPACES_COUNT: '5'
```

Before: User is able to create workspaces more than limit
After: Error is show when limit exceeded

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: multiuser/api/che-multiuser-api-resource/src/main/java/org/eclipse/che/multiuser/resource/api/workspace/LimitsCheckingWorkspaceManager.java [hunk 1]
```diff
@@ -117,6 +119,23 @@ public WorkspaceImpl createWorkspace(
...
+  @Override
+  @Traced
+  public WorkspaceImpl createWorkspace(
+      Devfile devfile,
+      String namespace,
+      Map<String, String> attributes,
+      FileContentProvider contentProvider)
+      throws ServerException, NotFoundException, ConflictException, ValidationException {
+    String accountId = accountManager.getByName(namespace).getId();
+    try (@SuppressWarnings("unused")
```
Comment (sparkoo) [inline multiuser/api/che-multiuser-api-resource/src/main/java/org/eclipse/che/multiuser/resource/api/workspace/LimitsCheckingWorkspaceManager.java:131]: I think you can name the variable `ignored` and you won't need SupressWarnings

**Review by sparkoo — APPROVED** [review]

**Comment by skabashnyuk** [comment]
[crw-ci-test --rebuild]

**Comment by skabashnyuk** [comment]
can we have a test for that?

**Comment by mshaposhnik** [comment]
All we can test is that `checkWorkspaceResourceAvailability()` is called, and it is problematic due to `super` call and locks are requires quite excessive mocking. This class is tested well in term of resourse checking validity.

**Review by skabashnyuk — APPROVED** [review]

**Pull request merged**